homeassistant/components/overkiz/cover_entities/vertical_cover.py

```diff
@@ -87,7 +87,7 @@ def current_cover_position(self) -> int | None:
         )
 
         # Uno devices can have a position not in 0 to 100 range when unknown
-        if position is None or position < 0 or position > 100:
+        if position is None or cast(int, position) < 0 or cast(int, position) > 100:
```

Comment:
Might be better to cast once as `int | None` or `Union[int,None]` directly above in `executor.select_state`
It will avoid having two casts here and another one below.

Reply:
I didn't know that this was possible! Will have a look and do that for all other cases.

Reply:
@epenet Instead of casting, what do you think if we create these kinds of method: `get_state_as_int`, `get_state_as_bool`, it will avoid casting and be more accurate.

Reply:
@tetienne I think that would be best yes, even better to do that in the backend library.
You'll need to decide what you do with the None values - it might be good to return `int | None`, `bool | None`, etc. in the library, and keep the `if None` checks inside HA.

Reply:
@epenet, the only difficulty is that we don't know this up front. At run-time, we get back a State object with a type (enum) that reflects a type. The reason why we can make the assumptions is because we know that the state we get by key, returns the type we need. (since it is tested and the API doesn't change).

Reply:
In the library, you could add `select_state_as_int` or `select_int_state` which does the cast.

```python
    @property
    def current_cover_position(self) -> int | None:
        ...
        position = self.executor.select_state_as_int(
            OverkizState.CORE_CLOSURE,
            OverkizState.CORE_CLOSURE_OR_ROCKER_POSITION,
            OverkizState.CORE_PEDESTRIAN_POSITION,
        )
```

Reply:
https://github.com/iMicknl/python-overkiz-api/pull/330

Reply:
Awesome! Shall we do this in a follow up PR? I think casting in this PR is fine, and we could refactor the whole integration + dep bump in a new PR.